proposals/threads/Overview.md

```diff
@@ -48,8 +48,48 @@ agent's cluster will have access to the additional linear memory.
 When a module has an imported linear memory, its data segments are copied into
 the linear memory when the module is instantiated.
 
-When the imported linear memory is shared, the writes are non-atomic and
-are not ordered.
+When linear memory is not shared, the ordering of data segment initialization is
+not observable. When linear memory is shared, it is possible for another module
```

Comment:
Perhaps I misunderstanding what you mean, but the JS.md doc is pretty explicit that there [cannot be a failure partway through intialization](https://github.com/WebAssembly/design/blob/master/JS.md#webassemblyinstance-objects):

> Segments may overlap and, if they do, the final value is the last value written in order. Note: there should be no errors possible that would cause this operation to fail partway through.